database/DevHome.Database/DatabaseModels/RepositoryManagement/Repository.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.EntityFrameworkCore;
+
+namespace DevHome.Database.DatabaseModels.RepositoryManagement;
+
+[Index(nameof(RepositoryName), nameof(RepositoryClonePath), IsUnique = true)]
+public class Repository
+{
+    public int RepositoryId { get; set; }
+
+    public string? RepositoryName { get; set; }
+
+    public string? RepositoryClonePath { get; set; }
+
+    // 1:1 relationship.  Repository is the parent and needs only
+    // the object of the dependant.
+    public RepositoryMetadata? RepositoryMetadata { get; set; }
```

Comment:
Technically, the metadata isn't a part of a repository.  A repository does not know if it is hidden or not.  "IsHidden" is used for DevHome.  

I modeled the schema after this idea.

You have a point.  From DevHomes perspective a repository comes with "IsHidden" because DevHome needs to know if the repository should be shown or not.

We have two ideologies.
1. Model the schema according to the objects and their relationships.
2. Model the schema according to the UI and how the data is grouped together.

I decided to go with the former.  If you have strong reasons for the latter, I want to hear them.

Reply:
What I mean here, is that if there's data about a repository that's in "RepositoryMetadata" and it is a 1:1 relationship, it should be a single table, not two tables that then need to be kept in sync or could get out of sync. Its more of helping the ORM understand the constraint and keeping the resulting data simple and consistent.